### DISCUSSION TITLE
Centralize vulnerability detection information on a single agent that sends it to the server.

### DISCUSSION THREAD

**Discussion by MathisLandreau** [wazuh/wazuh#25788] [Q&A]
Hello,
I need to do vulnerability detection of applications, OS,... on servers. But I can't put agents on each server for security reasons. I'd like to know if it's possible to upload the data that the agent sends to the server on a single machine so that it can send it back to the server. I know that it's possible with logs, but with information for vulnerability detection? how can I do it? 

The other solution I've found is to get the information from ansible. But I don't know if wazuh is able to retrieve this information and analyze it for automatic detection of version vulnerabilities...

Thanks for your help

**Comment by MiguelazoDS** [comment]
Hi @MathisLandreau, 

I think I understand your question, but let me know if I'm wrong. 

What you do NOT want to do is to register agents to a server, but you still want to know if a machine, without Wazuh agent installed, is vulnerable to some CVEs affecting OS or packages? Something similar to the agentless feature maybe? 

Is that correct?

**Reply by MathisLandreau** [reply]
Hi 
thank you for your answer but after that we chose to put agents ... 
But I have a question: is it possible to also detect updates of different software, not necessarily because of vulnerabilities, just an update?

**Comment by MiguelazoDS** [comment]
Hi @MathisLandreau. 

Could you provide an example, I didn't understand that. 

Do you mean to identify if, for example, the Firefox package is outdated? 

I'm afraid that is in charge of the package manager, but if you refer to Linux and the package is not installed by sources we can execute some command in the agent querying that information. For Windows, I don't see an easy way to do that. 

Take a look at this https://documentation.wazuh.com/current/user-manual/capabilities/command-monitoring/index.html

**Comment by sebasfalcone** [comment]
Inactive thread, please re-open it if necessary
